### ISSUE TITLE
Investigate the use of BGL for instance renames.

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#869]
Originally reported of Google Code with ID 813.

```
~# gnt-cluster --version
gnt-cluster (ganeti v2.10.1) 2.10.1

~# gnt-cluster version
Software version: 2.10.1
Internode protocol: 2100000
Configuration format: 2100000
OS api version: 20
Export interface: 0
VCS version: (ganeti) version v2.10.1

~# hspace --version
hspace (ganeti) version v2.10.1
compiled with ghc 6.12
running on linux x86_64

<b>What distribution are you using?</b>
Debian squeeze

After i sent an instance to be renamed, a migration entered the path
between the shutdown and the rename itself. Checking the needed locks,
it looks like the rename operation requires the Big Ganeti Lock
(cluster/BGL) to succeed. Is that the case? Could it be reduced to a
shared lock?

~# gnt-debug locks | grep 38341
cluster/BGL                                            shared
Jq6/Job38338/QUERY,Jq18/Job38336/I_CREATE shared:Jq20/Job38337/QUERY,
exclusive:Jq9/Job38341/I_RENAME,
shared:Jq1/Job38343/I_CREATE,Jq25/Job38342/C_VERIFY
```
Originally added on 2014-04-24 15:01:42 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2014-04-25 07:31:22 +0000 UTC.

Added Labels: Type-Defect Priority-High
Changed State: Accepted
Added to Milestone: Release2.13

**Comment by Ganeti-Issues-Migrator** [comment]
```
Fixed in:

8b3cd8ffc9400deeed381f541af76f0f809ff2eb Fix failing BGL whitelist test
771295610d4c3b9fcfff1b724cee63865bb7a3f8 Perform name resolution ASAP
14533030a16af190d613df7784df38d25b9c2a57 Remove BGL from LUInstanceRename
e1798d178dea0503325822df0a30367b9e56a9bb Factor out instance existence checks
9261433f5385f2dee65a8de7ce6f1baa0221c547 Allow coexistence of adding and acquiring locks
038a9f8f6c34d78e97cda16f913e267ce92d52f4 Allow _LockList to handle locking.ALL_SET

Can be cherry-picked into 2.12.
```
Originally added on 2014-07-18 17:54:57 +0000 UTC.

Changed State: Fixed

**Comment by Ganeti-Issues-Migrator** [comment]
```
As the fixes have been cherry-picked to 2.12 already, I'm changing the milestone.
```
Originally added on 2015-05-12 09:38:49 +0000 UTC.

Added to Milestone: Release2.12
Removed from Milestone: Release2.13

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2015-05-12 18:37:30 +0000 UTC.

Changed State: Released
